src/plugins/charts/public/services/colors/mapped_colors.ts

```diff
@@ -100,7 +100,13 @@ export class MappedColors {
 
     // Generate a color palette big enough that all new keys can have unique color values
     const allColors = _(this._mapping).values().union(configColors).union(oldColors).value();
-    const colorPalette = createColorPalette(allColors.length + keysToMap.length);
+    const numColors = allColors.length + keysToMap.length;
+    const colorPalette = euiPaletteColorBlind({
+      rotations: Math.ceil(numColors / 10),
+      direction: 'both',
+      order: 'group',
```

Comment:
I think we actually want to use 
```suggestion
      order: 'append',
```

The idea is that the first 10 colors assigned should be the default (non-rotated). Then the next 10 (11-20) come from one of the first rotations, in the same order. Then the next 10 (21-30) come from the first rotation in the other direction. I think we may also want to remove the `natural` sort to get them ordered correctly.

Reply:
Yes, I think the screenshot was misleading before. I've updated the screenshot after adjusting the order and sort criteria.